presto-docs/src/main/sphinx/sql/select.rst

```diff
@@ -92,6 +92,95 @@ Additionally, the relations within a ``WITH`` clause can chain::
     relation is used. This means that if the relation is used more than once and the query
     is non-deterministic, the results may be different each time.
 
+WITH RECURSIVE Clause
+---------------------
+
+The ``WITH RECURSIVE`` clause is a variant of the ``WITH`` clause. It can define
+a list of queries to process, including recursive processing of suitable
+queries.
+
+.. warning::
+
+    This feature is experimental only. Proceed to use it only if you understand
+    potential query failures and the impact of the recursion processing on your
+    workload.
+
+.. code-block:: none
+
+   WITH RECURSIVE T(n) AS (
+    <recursion base relation>
+    UNION (ALL | DISTINCT)
+    <recursion step relation: SELECT  FROM ... T...>
+    )
+    ... SELECT ... FROM T
```

Comment:
I think we don't need this generic example. The other example + explanation is enough.

Reply:
ok .. removed